### PR TITLE
Initial patch fixing multi-column partition pruning

### PR DESCRIPTION
**Pull request by soubhik-c**
## Changes proposed in this pull request

Verifying the number of partitioning columns and filter expressions covering them exists. Removed Sample table pruning because sampling itself contains lesser data. All tests are passing now.

## Patch testing

precheckin in progress, no new failures observed yet.

## ReleaseNotes.txt changes

(Does this change require an entry in ReleaseNotes.txt? If yes, has it been added to it?)

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)

### CODE REVIEW

File: cluster/src/test/scala/io/snappydata/cluster/QueryRoutingSingleNodeSuite.scala [hunk 1]
```diff
@@ -292,7 +292,7 @@ class QueryRoutingSingleNodeSuite extends SnappyFunSuite with BeforeAndAfterAll
       insertRows(tableName2, 1000, serverHostPort)
       // As part of fix to SNAP-1478 thie below query should be enabled
       // and verified.
-      // query1(tableName1, tableName2, serverHostPort)
```
Comment (vivekwiz) [inline cluster/src/test/scala/io/snappydata/cluster/QueryRoutingSingleNodeSuite.scala]: Even above lines should go

File: cluster/src/test/scala/io/snappydata/cluster/PreparedQueryRoutingSingleNodeSuite.scala [hunk 2]
```diff
@@ -483,7 +483,7 @@ class PreparedQueryRoutingSingleNodeSuite extends SnappyFunSuite with BeforeAndA
       query1(tableName1, tableName2)
       query2(tableName1, tableName2)
       query3(tableName1, tableName2)
-      // query4(tableName1, tableName2) // TODO - Enable with SNAP-1478
+      query4(tableName1, tableName2) // TODO - Enable with SNAP-1478
```
Comment (vivekwiz) [inline cluster/src/test/scala/io/snappydata/cluster/PreparedQueryRoutingSingleNodeSuite.scala]: Please remove comments.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala [hunk 3]
```diff
@@ -179,7 +181,6 @@ abstract class BaseColumnFormatRelation(
       Seq[RDD[InternalRow]]) = {
     val rdd = scanTable(table, requiredColumns, filters, -1)
     val partitionEvaluator = rdd match {
-      case c: ColumnarStorePartitionedRDD => c.getPartitionEvaluator
       case r => () => r.partitions
     }
```
Comment (vivekwiz) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: Can we remove partitionEvaluator and replace it with "() => rdd.partitions" in usage?

File: cluster/src/test/scala/io/snappydata/cluster/QueryRoutingSingleNodeSuite.scala (hunk 1 above)
Comment (vivekwiz) [inline cluster/src/test/scala/io/snappydata/cluster/QueryRoutingSingleNodeSuite.scala]: Please also remove couple of lines for TODO above declaration of query1

File: cluster/src/test/scala/io/snappydata/cluster/PreparedQueryRoutingSingleNodeSuite.scala (hunk 2 above)
Comment (soubhik-c) [inline cluster/src/test/scala/io/snappydata/cluster/PreparedQueryRoutingSingleNodeSuite.scala]: done.

File: core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala (hunk 3 above)
Comment (soubhik-c) [inline core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala]: yup. done.

**Pull request merged**